File: IHaveATheory/app/src/main/java/com/avoid/ihaveatheory/util/SaveFileHandler.java
package com.avoid.ihaveatheory.util;


import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import com.avoid.ihaveatheory.global.Session;
import com.avoid.ihaveatheory.model.SaveFile;
import com.google.gson.Gson;

public class SaveFileHandler {
    private SharedPreferences preferences;
    private Gson gson;

    public SaveFileHandler(Context context) {
        preferences = context.getSharedPreferences("IHaveATheoryPrefs", 0);
        gson = new Gson();
    }

    public void newGame() {
        preferences.edit().remove("save_file").commit();
        Session.currentSaveFile = new SaveFile();
    }

    public void loadGame() {
        String json = preferences.getString("save_file", "");
        Session.currentSaveFile = gson.fromJson(json, SaveFile.class);
    }

    public void saveGame(){
        SharedPreferences.Editor prefsEditor = preferences.edit();
        String json = gson.toJson(Session.currentSaveFile);
        prefsEditor.putString("save_file", json);
        prefsEditor.commit();

        Log.d(">>>>>>>>>>>>>>>>",json);
    }

    public boolean saveExists(){
        String json = preferences.getString("save_file", "");

        return !json.equals("");
    }
}

File: app/src/main/java/com/avoid/ihaveatheory/util/SaveFileHandler.kt
package com.avoid.ihaveatheory.util


import android.content.Context
import com.avoid.ihaveatheory.global.Session
import com.avoid.ihaveatheory.model.SaveFile
import java.io.*

class SaveFileHandler(private val context: Context) {
    private var fileOutputStream: FileOutputStream? = null
    private var objectOutputStream: ObjectOutputStream? = null

    private var fileInputStream: FileInputStream? = null
    private var objectInputStream: ObjectInputStream? = null

    fun newGame() {
        Session.currentSaveFile = SaveFile()
    }

    fun loadGame() {
        try {
            fileInputStream = context.openFileInput("saveFile")
            objectInputStream = ObjectInputStream(fileInputStream)

            Session.currentSaveFile = objectInputStream!!.readObject() as SaveFile

            objectInputStream!!.close()
            fileInputStream!!.close()
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    fun saveGame() {
        try {
            fileOutputStream = context.openFileOutput("saveFile", Context.MODE_PRIVATE)
            objectOutputStream = ObjectOutputStream(fileOutputStream)

            objectOutputStream!!.writeObject(Session.currentSaveFile)

            objectOutputStream!!.close()
            fileOutputStream!!.close()
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    fun saveExists(): Boolean {
        try {
            fileInputStream = context.openFileInput("saveFile")
            return fileInputStream != null
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
            return false
        }

    }
}
